x-pack/plugins/synthetics/server/legacy_uptime/lib/saved_objects/migrations/monitors/8.8.0.ts

```diff
@@ -118,3 +126,102 @@ const omitZipUrlFields = (fields: BrowserFields) => {
 
   return formatSecrets(validationResult.decodedMonitor);
 };
+
+const updateThrottlingFields = (
+  doc: SavedObjectUnsanitizedDoc<
+    SyntheticsMonitorWithSecrets &
+      Partial<{
+        [LegacyConfigKey.THROTTLING_CONFIG]: string;
+        [LegacyConfigKey.IS_THROTTLING_ENABLED]: boolean;
+        [LegacyConfigKey.DOWNLOAD_SPEED]: string;
+        [LegacyConfigKey.UPLOAD_SPEED]: string;
+        [LegacyConfigKey.LATENCY]: string;
+      }>
+  >,
+  logger: SavedObjectMigrationContext
+) => {
+  try {
+    const { attributes } = doc;
+
+    const migrated = {
+      ...doc,
+      attributes: {
+        ...doc.attributes,
+        [ConfigKey.CONFIG_HASH]: '',
+      },
+    };
+
+    const isThrottlingEnabled = attributes[LegacyConfigKey.IS_THROTTLING_ENABLED];
+    if (isThrottlingEnabled) {
+      const download = String(attributes[LegacyConfigKey.DOWNLOAD_SPEED])!;
```

Comment:
My original comment wasn't addressed.

I mentioned before 
```
If for whatever reason attributes[LegacyConfigKey.DOWNLOAD_SPEED] is undefined, maybe we should do || defaultDownLoadSpeed, etc. for all the below.
```

That way we don't have to use `!` and get the benefit of more defensive coding.